tools/clang/lib/SPIRV/LowerTypeVisitor.cpp

```diff
@@ -761,6 +761,26 @@ const SpirvType *LowerTypeVisitor::lowerResourceType(QualType type,
     return valType;
   }
 
+  if (name == "ConstantBuffer") {
+    // ConstantBuffer<T> is lowered as T
+    if (rule == SpirvLayoutRule::Void) {
+      rule = getCodeGenOptions().sBufferLayoutRule;
+    }
```

Comment:
I don't think you need this. This will cause the type to get the offset and other decorations. They are not needed for the local type. This is done for some other resource types because their "local" version are actually pointers to the global type, so the rule is change to get the correct type for the pointee.

Reply:
I believe now that this is being used to generate both the extern variable and the local ones that it's best to set this as `cBufferLayoutRule` or `tBufferLayoutRule` according to the buffer type, so that only one type is generated.